torch/testing/_internal/common_utils.py

```diff
@@ -135,6 +136,87 @@ def wrapper(*args, **kwargs):
         fn(*args, **kwargs)
     return wrapper
 
+# Tries to extract the test function and full test ID of the current function under test
+# by crawling the stack. If unsuccessful, returns a pair of Nones.
+def extract_test_fn_and_id() -> Optional[Tuple[Callable, str]]:
```

Comment:
nit: should the return type be `Tuple[Optional[Callable], Optional[str]]`?

Reply:
yep sure enough!